config/src/lib.rs

```diff
@@ -1241,6 +1245,10 @@ impl Config {
     }
 
     pub fn apply_cmd_defaults(&self, cmd: &mut CommandBuilder) {
+        if let Some(ref cwd) = self.default_cwd {
```

Comment:
I'm a little "worried" about how this interacts with other things that want to set the cwd.
There are a couple of places that come to mind:

 * The logic that applies the OSC 7 specified cwd to the command; `LocalDomain::spawn` and `LocalDomain::split_pane`.  Those apply the cwd *after* calling `apply_cmd_defaults` so those seem ok
 * `wezterm start --cwd /some/where` which pre-populates the cwd before `spawn_tab_in_default_domain_if_mux_is_empty` calls `spawn` which calls down to `apply_cmd_defaults` here.
 * Specifying a `cwd` setting in a launch menu item (see: https://wezfurlong.org/wezterm/config/launch.html#the-launcher-menu).  I didn't check to see if this would behave correctly or not; would be great if you could verify that!

I think the `--cwd` option should override the value specified in the configuration file, so what I think needs to be done is to introduce:

```rust
   pub fn get_cwd(&self) -> Option<&OsString> {
       self.cwd.as_ref()
   }   
```

to the `CommandBuilder` in `pty/src/cmdbuilder.rs`

so that the logic here in this function can be something like:

```rust
    if let Some(ref cwd) = self.default_cwd {
       // Allow `--cwd` option to take precedence
       if cmd.get_cwd().is_none() {
          cmd.cwd(cwd);
       }
    }
```

Reply:
For the last bit of logic there you suggest, do you have any feelings about me writing it like this?

```rust
// Apply `default_cwd` only if `cwd` is not already set, allows `--cwd`
// option to take precedence
if let (None, Some(cwd)) = (cmd.get_cwd(), &self.default_cwd) {
    cmd.cwd(cwd);
}
```

Reply:
Go for it!